File: Commons/Action/src/main/java/com/greentree/commons/action/observer/integer/ObjIntAction.java
package com.greentree.commons.action.observer.integer;

import java.util.function.ObjIntConsumer;

import com.greentree.commons.action.MultiAction;
import com.greentree.commons.action.observable.ObjIntObservable;
import com.greentree.commons.action.observer.ObjIntObserver;

public final class ObjIntAction<T> extends MultiAction<ObjIntConsumer<? super T>>
		implements ObjIntObservable<T>, ObjIntObserver<T> {
	
	private static final long serialVersionUID = 1L;
	
	@Override
	public void event(T e, int i) {
		for(var l : listeners)
			l.accept(e, i);
	}
	
}

File: Commons/Action/src/main/java/com/greentree/commons/action/observer/integer/ObjIntAction.kt
package com.greentree.commons.action.observer.integer

import com.greentree.commons.action.MultiAction
import com.greentree.commons.action.observable.ObjIntObservable
import com.greentree.commons.action.observer.ObjIntObserver
import java.util.function.ObjIntConsumer

class ObjIntAction<T> : MultiAction<ObjIntConsumer<in T>>(), ObjIntObservable<T>, ObjIntObserver<T> {

	override val listenerCount: Int
		get() = listeners.size()

	override fun event(e: T, i: Int) {
		for(l in listeners)
			l.accept(e, i)
	}
}